static/js/util/profile_edit.js

```diff
@@ -81,3 +104,44 @@ export function saveAndContinue(next) {
     browserHistory.push(next)
   });
 }
+
+export function editProfileObjectArray (arrayName, blankEntry, formCallback) {
+  const { updateProfile, profile } = this.props;
+  if ( profile[arrayName] ) {
+    let editForms = profile[arrayName].map( (entry, index) => (
+      formCallback.bind(this)(entry, index)
+    ));
+    let addAnotherBlankEntry = () => {
+      let clone = Object.assign({}, profile);
+      clone[arrayName].push(Object.assign({}, blankEntry));
```

Comment:
Since `Object.assign` only does a shallow copy, can you do this instead:

    clone[arrayName] = clone[arrayName].concat(blankEntry);

Since `blankEntry` isn't changed here you shouldn't need to clone it